#!/usr/bin/env python
PACKAGE = "to_qnx"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("vel",		double_t, 0, "velocity",           0, 0, 5);
gen.add("mode",		int_t, 0, "mode",   		   1, 0, 2);

# desired PID parameter for Steer
gen.add("steer_p",      double_t, 0, "Steeing P",              0.5, 0, 5);
gen.add("steer_i",      double_t, 0, "Steeing I",              0.01, 0, 1);
gen.add("steer_d",      double_t, 0, "Steeing D",              0, 0, 1);

# desired PID parameter for Velociy
gen.add("accel_p",      double_t, 0, "accel P",              2, 0, 5);
gen.add("accel_i",      double_t, 0, "accel I",              0.5, 0, 10);
gen.add("accel_d",      double_t, 0, "accel D",              10, 0, 10);

# reset QNX
gen.add("reset",        int_t, 0, "reset_QNX",                   0, 0, 1)

gen.add("current_vel",  double_t, 0, "current velocity",         0, 0, 10);

exit(gen.generate(PACKAGE, "to_qnx", "qnx_command"))

